riverdriver/riverpgxv5/migration/main/005_river_migration_add_line.up.sql

```diff
@@ -15,4 +15,35 @@ INSERT INTO river_migration
 SELECT created_at, 'main', version
 FROM river_migration_old;
 
-DROP TABLE river_migration_old;
\ No newline at end of file
+DROP TABLE river_migration_old;
+
+--
+-- Create `river_client` and derivative.
+--
+-- This feature hasn't quite yet been implemented, but we're taking advantage of
+-- the migration to add the schema early so that we can add it later without an
+-- additional migration.
+--
+
+CREATE UNLOGGED TABLE river_client (
+    name text PRIMARY KEY NOT NULL,
+    created_at timestamptz NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    metadata jsonb NOT NULL DEFAULT '{}' ::jsonb,
+    paused_at timestamptz,
+    updated_at timestamptz NOT NULL,
+    CONSTRAINT name_length CHECK (char_length(name) > 0 AND char_length(name) < 128)
+);
+
+-- Differs from `river_queue` in that it tracks the queue state for a particular
+-- active client.
+CREATE UNLOGGED TABLE river_client_queue (
+    river_client_name text NOT NULL REFERENCES river_client (name) ON DELETE CASCADE,
+    name text NOT NULL,
+    created_at timestamptz NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    max_workers bigint NOT NULL DEFAULT 0,
+    num_jobs_completed bigint NOT NULL DEFAULT 0,
+    num_jobs_running bigint NOT NULL DEFAULT 0,
```

Comment:
While these are named very similarly, they seem to have very different purposes (if I'm understanding it correctly). One is a historical aggregate stat of how many jobs the client has worked on that queue, while the other is showing a real-time count of in-progress jobs. Almost wondering if we should name it more similarly to `max_workers`.

Minor though, I don't feel strongly about it.

Reply:
Hmm, I kind of feel like it's pretty descriptive and I'm having a hard time thinking of a better name. Did you have one in mind?